metrics/server.go

```diff
@@ -0,0 +1,73 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/rs/zerolog"
+)
+
+// Server is the http server that will be serving the /metrics request for prometheus
+type Server struct {
+	server *http.Server
+	log    zerolog.Logger
+}
+
+// NewServer creates a new server that will start on the specified port,
+// and responds to only the `/metrics` endpoint
+func NewServer(log zerolog.Logger, port uint) *Server {
+	addr := ":" + strconv.Itoa(int(port))
+
+	mux := http.NewServeMux()
+	endpoint := "/metrics"
+	mux.Handle(endpoint, promhttp.Handler())
+	log.Info().Str("address", addr).Str("endpoint", endpoint).Msg("metrics server started")
+
+	server := &Server{
+		server: &http.Server{Addr: addr, Handler: mux},
+		log:    log,
+	}
+
+	return server
```

Comment:
```suggestion
return &Server{
		server: &http.Server{Addr: addr, Handler: mux},
		log:    log,
	}
```

Reply:
I log server info after creating it in new commit